mat/cholesky.go

```diff
@@ -1065,12 +1066,73 @@ func (c *PivotedCholesky) Rank() int {
 
 // Cond returns the condition number of the factorized matrix.
 func (c *PivotedCholesky) Cond() float64 {
-	if !c.ok {
+	if c.chol == nil {
 		panic(badCholesky)
 	}
 	return c.cond
 }
 
+// RawU returns the Triangular matrix used to store the Cholesky factorization
+// of the original matrix A. If the returned matrix is modified, the
+// factorization is invalid and should not be used.
+//
+// If Factorized returned false, the rows of U from Rank to n will contain zeros
+// and so U will be upper trapezoidal.
+//
+// The returned value is usable only if Factorize has been called.
+func (c *PivotedCholesky) RawU() Triangular {
+	return c.chol
+}
+
+// UTo stores the n×n upper triangular matrix U from the Cholesky factorization
+//
+//	Pᵀ * A * P = Uᵀ * U.
+//
+// into dst. If dst is empty, it is resized to be an n×n upper triangular
+// matrix. When dst is non-empty, UTo panics if dst is not n×n or not Upper.
+//
+// If Factorized returned false, the rows of U from Rank to n will contain zeros
+// and so U will be upper trapezoidal.
+func (c *PivotedCholesky) UTo(dst *TriDense) {
+	if c.chol == nil {
+		panic(badCholesky)
+	}
+	n := c.chol.mat.N
+	if dst.IsEmpty() {
+		dst.ReuseAsTri(n, Upper)
+	} else {
+		n2, kind := dst.Triangle()
+		if n != n2 {
+			panic(ErrShape)
+		}
+		if kind != Upper {
+			panic(ErrTriangle)
+		}
+	}
+	dst.Copy(c.chol)
+}
+
+// Pivots returns the column permutation p that represents the permutation
+// matrix P from the Cholesky factorization
+//
+//	Pᵀ * A * P = Uᵀ * U
+//
+// such that the nonzero entries are P[p[k],k] = 1.
+func (c *PivotedCholesky) Pivots(dst []int) []int {
```

Comment:
In `LU` this function name is in singular and I don't know if it's better to rename this to be consistent with a bad name, rename in LU to plural but break the API, or rename both, in LU to `RowPivots` and here to `ColumnPivots`.

Reply:
I think probably the last option.

Reply:
I've renamed Pivots here to ColumnPivots. I'll deprecate LU.Pivot and add RowPivots in a separate PR.